java-compiler-testing/src/main/java/io/github/ascopes/jct/junit/JavacCompilersProvider.java

```diff
@@ -26,18 +25,14 @@
  * @author Ashley Scopes
  * @since 0.0.1
  */
-public final class JavacCompilersProvider extends AbstractCompilersProvider
+final class JavacCompilersProvider extends AbstractCompilersProvider
     implements AnnotationConsumer<JavacCompilerTest> {
 
   /**
    * Initialise the provider.
-   *
-   * <p>This is only visible for testing purposes, users should have no need to
-   * initialise this class directly.
    */
-  @VisibleForTestingOnly
-  public JavacCompilersProvider() {
-    // Visible for testing only.
+  JavacCompilersProvider() {
```

Comment:
technically a breaking change but the docs specifically tell users not to use it, so I am not changing the major version for this.

Likewise, moving the class to package private scope should not constitute an actual breaking change as this "do not use" constructor would be the only way to initialize an instance of the class... therefore this class itself shouldn't be used by users either.